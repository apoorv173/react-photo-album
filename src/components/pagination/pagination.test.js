import React from "react";
import Pagination from "./pagination";
import { shallow, mount } from "enzyme";

describe("Pagination comnponent", () => {

    const onPrevClick = jest.fn();
    const onNextClick = jest.fn();
    it("Should render the Pagination with both the buttons enabled", () => {
        const tree = shallow(<Pagination onPrevClick={onPrevClick} onNextClick={onNextClick} ></Pagination>)
        expect(tree).toMatchSnapshot();
    });

    it("Should render the Pagination with both the buttons disabled", () => {
        const tree = shallow(<Pagination 
                onPrevClick={onPrevClick} 
                onNextClick={onNextClick} 
                prevDisabled={true}
                nextDisabled={true}
            ></Pagination>)
        expect(tree).toMatchSnapshot();
    });

    it('should render all the components isnide the pagination component', () => {
        const wrapper = mount(<Pagination onPrevClick={onPrevClick} onNextClick={onNextClick} ></Pagination>);
        expect(wrapper.find('.pagination').length).toBe(1);
        expect(wrapper.find('.left').length).toBe(1);
        expect(wrapper.find('.right').length).toBe(1);
    });

    it('should call the on on previous and on next event', () => {
        const wrapper = mount(<Pagination onPrevClick={onPrevClick} onNextClick={onNextClick}></Pagination>);

        wrapper.find(".prev-btn").simulate("click");
        expect(onPrevClick).toHaveBeenCalled();

        wrapper.find(".next-btn").simulate("click");
        expect(onNextClick).toHaveBeenCalled();
    });
});