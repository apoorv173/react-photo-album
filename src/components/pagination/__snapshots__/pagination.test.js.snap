// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination comnponent Should render the Pagination with both the buttons disabled 1`] = `ShallowWrapper {}`;

exports[`Pagination comnponent Should render the Pagination with both the buttons enabled 1`] = `ShallowWrapper {}`;

exports[`Pagination comnponent Should render the modal window with both the buttons disabled 1`] = `ShallowWrapper {}`;

exports[`Pagination comnponent Should render the modal window with both the buttons enabled 1`] = `ShallowWrapper {}`;
